src/main/java/org/prebid/server/bidder/videobyte/VideobyteBidder.java

```diff
@@ -0,0 +1,155 @@
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<BidRequest>> requests = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpVideobyte extImpVideobyte = parseImpExt(imp);
+                requests.add(createRequest(request, imp, extImpVideobyte));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        return Result.of(requests, errors);
+    }
+
+    private ExtImpVideobyte parseImpExt(Imp imp) throws PreBidException {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), VIDEOBYTE_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            final String errorMessage = "Ignoring imp id=%s, error while decoding, err: %s";
+            throw new PreBidException(String.format(errorMessage, imp.getId(), e.getMessage()));
+        }
+    }
+
+    private HttpRequest<BidRequest> createRequest(BidRequest bidRequest, Imp imp, ExtImpVideobyte extImpVideobyte)
+            throws PreBidException {
+
+        final BidRequest modifiedBidRequest = bidRequest.toBuilder().imp(Collections.singletonList(imp)).build();
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(createUri(extImpVideobyte))
+                .headers(headers(modifiedBidRequest))
+                .body(mapper.encodeToBytes(modifiedBidRequest))
+                .payload(modifiedBidRequest)
+                .build();
+    }
+
+    private String createUri(ExtImpVideobyte extImpVideobyte) throws PreBidException {
+        try {
+            final URIBuilder uriBuilder = new URIBuilder(endpointUrl)
+                    .addParameter("source", "pbs")
+                    .addParameter("pid", extImpVideobyte.getPublisherId());
+            letIfNotBlank(extImpVideobyte.getPlacementId(), it -> uriBuilder.addParameter("placementId", it));
+            letIfNotBlank(extImpVideobyte.getNetworkId(), it -> uriBuilder.addParameter("nid", it));
+
+            return uriBuilder.build().toString();
+        } catch (URISyntaxException e) { // shouldn't really happen
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private static MultiMap headers(BidRequest bidRequest) {
+        final MultiMap headers = HttpUtil.headers();
+
+        final Site site = bidRequest.getSite();
+        if (site != null) {
+            letIfNotBlank(site.getDomain(), it -> headers.add(HttpUtil.ORIGIN_HEADER, it));
+            letIfNotBlank(site.getRef(), it -> headers.add(HttpUtil.REFERER_HEADER, it));
```

Comment:
Can be replaced with something like:
 `HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.ORIGIN_HEADER, site.getDomain())`